lib/Doctrine/CouchDB/HTTP/MultipartParserAndSender.php

```diff
@@ -0,0 +1,416 @@
+<?php
+
+
+namespace Doctrine\CouchDB\HTTP;
+
+
+/**
+ * Streams the multipart data from the source to the target and thus makes the
+ * transfer with lesser memory footprint.
+ *
+ * Class MultipartParserAndSender
+ * @package Doctrine\CouchDB\HTTP
+ */
+class MultipartParserAndSender
```

Comment:
This class needs tests, just the one integration test is not enough to test all the many edge cases we have here.